worker/src/lib/dbLogger/DBLoggable.ts

```diff
@@ -605,12 +634,42 @@ export class DBLoggable {
 
     // If no data or error, return
     if (!responseResult.data || responseResult.error) {
+      const body = {
+        helicone_error: "error getting response, " + responseResult.error,
+        helicone_repsonse_body_as_string: (
+          await this.response.getResponseBody()
+        ).body,
+      };
+
+      const s3Result = await db.s3Client.storeRequestResponse(
+        authParams.organizationId,
+        this.request.requestId,
+        JSON.stringify(requestResult.data.body),
+        JSON.stringify(body)
+      );
+
+      if (s3Result.error) {
+        return err(s3Result.error);
+      }
+
       return responseResult;
     }
 
+    const s3Result = await db.s3Client.storeRequestResponse(
+      authParams.organizationId,
+      this.request.requestId,
+      requestResult.data.body,
+      responseResult.data.body
+    );
+
+    if (s3Result.error) {
+      console.log("Error storing request response", s3Result.error);
+      return err(s3Result.error);
+    }
```

Comment:
Also I dont think we should return here. Just incase

Reply:
for now...

Reply:
updated